\section{Datasets}
\label{sec:datasets}
This sections describes datasets used in this paper. Datasets can be devided into two groups.
The first one includes datasets from TUDatasets \cite{TUDataset}, which is
a large collection of graph datasets, introduced in \cite{TUDataset} and very common and widely used for benchmarking of GNN models.
In current paper there are used bioinformatics datsets containing networks representin micro- and macromolecules: \texttt{MUTAG}, \texttt{PROTEINS},
\texttt{ENZYMES}, \texttt{DD}. Properties of datasets are shown in table \ref{tab:datasets}
\begin{center}
    \begin{table}[h]
        \scriptsize
        \begin{tabular}{|l|l|l|l|l|l|l|l|}
            \hline
                                 & MUTAG        & PROTEINS     & ENZYMES             & DD            & Brain fMRI & Kidney RNA Seq \\ \hline
            \# Graphs            & 188          & 1113         & 600                 & 1778          & 124        & 299            \\ \hline
            \# Classes           & 2            & 2            & 6                   & 2             & 2          & 3              \\ \hline
            \# Samples per class & 63/125       & 663/450      & 100/100/100/100/100 & 691/487       & 70/54      & 159/90/50      \\ \hline
            \# Vertices          & $\approx 18$ & $\approx 39$ & $\approx 33$        & $\approx 284$ & 263        & 1034           \\ \hline
            Node features        & 7            & 3            & 3                   & 89            & 0          & 0              \\ \hline
            Max node degree      & 4            & 25           & 9                   & 19            & 238        & 105            \\ \hline
            Max diameter         & 15           & 64           & 37                  & 83            & 2          & 7              \\ \hline
            Has edge weights     & no           & no           & no                  & no            & yes        & yes            \\ \hline
            Has edge attrubutes  & yes          & no           & no                  & no            & no         & no             \\ \hline
        \end{tabular}
    \caption{Datasets properties}
    \label{tab:datasets}
    \end{table}
\end{center}
\texttt{MUTAG} dataset \cite{MUTAG} contains 188 graphs representing chemical nitroaromatic compounds.
Vertices af graphs respresent atoms, labeled by one-hot encoded atom type.
Edges represent bonds between the corresponding atoms correponds to their mutagenicity on Salmonella typhimurium.

Grpoup of \texttt{PROTEINS}, \texttt{ENZYMES}, \texttt{DD} datasets contains networks
representng macromolecules of proteins. Graphs from \texttt{PROTEINS} and \texttt{ENZYMES} datasets
are based on  graph model for proteins introduced in \cite{ProteinBased}.
In that graph model nodes represent secondary structure elements and are annotated by their types.
Nodes are connected by the edge if they are neighbors along the amino acid sequence or one of three nearest neighbors in
space. Using this approach \texttt{ENZYMES} was derived from BRENDA (BRaunschweig ENzyme DAtabase)\cite{Brenda} dataset, which
contains a large collection of enzyme and metabolic information, based on primary literature. Each graph in \texttt{ENZYMES}
has one of six classes, which reflects the catalyzed chemical reaction (6 classes). It contains 600 graphs. Dataset \texttt{PROTEINS} containing
1113 graphs, was derived  from abother dataset, presented in \cite{ProtBase2}. The task for \texttt{PROTEINS} dataset is binary classification:
whether a protein is an enzyme. Dataset \cite{DD} contains 1178 protein structures having nodes representing
individual amino acids and edges their spatial proximity.

Another group consists of two real medical datasets used for benchmarking \texttt{Netpro2vec} model in \cite{Netpro2vec}: \texttt{Brain fMRI} and
\texttt{Kidney RNASeq} datasets.

\texttt{Brain fMRI} dataset includes 124 real graphs derived in \cite{BrainFmri} from the Center for Biomedical Research Excellence (COBRE) dataset, which contains
functional magnetic resonance imaging (fMRI) time-series data. Brain fMRI dataset includes 124 graphs: 54 graphs from Schizophrenia subjects (Sch) and 70 graphs from
healthy controls (Ctrl). Each graph contains 263 nodes corresponding to different brain regions. The edges have weights representing Fisher-transformed correlation between
the fMRI time-series of the nodes after ranking.

\texttt{Kidney RNASeq} dataset consists of metabolic networks constructed by mapping gene expression data on the biochemical reactions
extracted from the kidney tissue metabolic model. Gene expression data is obtained from TCGA-KIRC and TCGA-KIRP projects
of the Genomic Data Commons \footnote{\url{https://portal.gdc.cancer.gov}} portal. Datasets was prepared by authors of \cite{Netpro2vec}
Kidney RNASeq dataset contains 299 networks devided into three classes. Two of them corresponds to the two most common types(according to National Cancer Institute \footnote{\url{https://www.cancer.gov/pediatric-adult-rare-tumor/rare-tumors/rare-kidney-tumors/clear-cell-renal-cell-carcinoma}})
of kidney cancer: Clear Cell Renal Cell Carcinoma (159 samples) and Papillary Renal Cell Carcinoma (90 samples). The third classs Solid Tissue Normal represents the absence of pathology
and contains 50 samples.

Both datasets were downloaded from cbs-group repository\footnote{\url{https://github.com/cds-group/GraphDatasets}} \footnote{\url{https://github.com/cds-group/Netpro2vec}}, authors of Netpro2vec model \cite{Netpro2vec}.
